modules/dockerregistry/dockerregistry.go

```diff
@@ -0,0 +1,45 @@
+package dockerregistry
+
+import (
+	"context"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+// DockerRegistryContainer represents the DockerRegistry container type used in the module
+type DockerRegistryContainer struct {
+	testcontainers.Container
+}
+
+// RunContainer creates an instance of the DockerRegistry container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*DockerRegistryContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        "docker.io/registry:latest",
+		ExposedPorts: []string{"5000/tcp"},
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		opt.Customize(&genericContainerReq)
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DockerRegistryContainer{Container: container}, nil
+}
+
+func WithAuthentication(dockerAuthConfig string) testcontainers.CustomizeRequestOption {
```

Comment:
Let's define the expectations on how to use this module: that would a user expect if the registry runs with authentication?
- should it be possible to pass an arbitrary `htpassw` file? See https://github.com/testcontainers/testcontainers-go/blob/main/testdata/auth/htpasswd
- should it be possible to load/push images to it?

I'd try to start with a container request like the one in this test: https://github.com/testcontainers/testcontainers-go/blob/main/docker_auth_test.go#L275-L299, allowing customising the auth file.

Then, the test for the module should check that's is posible to log in into that registry.

Does it make sense to you?

Reply:
Yes it is making sense to me.

Do we want also a realm as a parameter?

Reply:
I do not have a strong opinion... I think we can start simple, and iterate with a follow-up PR for the realm if you want. Or if you see it important, add it here. What is your preference? At the same time, if you consider it important for users of the module, then we can include it since the beginning.